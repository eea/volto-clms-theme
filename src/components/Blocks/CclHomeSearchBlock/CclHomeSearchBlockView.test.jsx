import React from 'react';
import renderer from 'react-test-renderer';
import CclSearchBlockView from './CclHomeSearchBlockView';
import CclSearchBlockEdit from './CclHomeSearchBlockEdit';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';

const mockSearchBlockPros = {
  title: 'Test title!',
  link: [{ '@id': '#' }],
};

describe('CclSearchBlock', () => {
  it('Check search block view', () => {
    const SearchBlockView = renderer
      .create(
        <MemoryRouter>
          <IntlProvider locale="en">
            <CclSearchBlockView data={mockSearchBlockPros}>
              <p>Search block view test</p>
            </CclSearchBlockView>
          </IntlProvider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(SearchBlockView).toMatchSnapshot();
  });
  it('Check search block edit', () => {
    const SearchBlockEdit = renderer
      .create(
        <MemoryRouter>
          <IntlProvider locale="en">
            <CclSearchBlockEdit data={mockSearchBlockPros}>
              <p>Search block edit test</p>
            </CclSearchBlockEdit>
          </IntlProvider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(SearchBlockEdit).toMatchSnapshot();
  });
});
