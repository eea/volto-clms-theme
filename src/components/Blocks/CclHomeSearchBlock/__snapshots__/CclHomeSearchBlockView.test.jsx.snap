// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CclSearchBlock Check search block edit 1`] = `
<div
  className="home-datasets-container"
>
  <div
    className="ccl-container"
  >
    <h3>
      Find and download data
    </h3>
    <div
      className="home-datasets-search"
    >
      <form
        className="ccl-form search-form"
      >
        <input
          aria-label="Search datasets"
          className="ccl-text-input"
          id="home_search_datasets"
          name="homeSearchDatasets"
          placeholder="Search datasets"
          type="text"
        />
        <button
          aria-label="Search"
          className="ccl-button"
          type="submit"
        >
          <span
            className="ccl-icon-zoom"
          />
        </button>
      </form>
    </div>
    <div
      className="home-datasets-text"
    >
      <span>
        More than XXX.XXX datasets in our 
      </span>
      <a
        href="./dataset-catalogue.html"
      >
        Dataset catalogue
      </a>
    </div>
  </div>
</div>
`;

exports[`CclSearchBlock Check search block view 1`] = `
<div
  className="home-datasets-container"
>
  <div
    className="ccl-container"
  >
    <h3>
      Find and download data
    </h3>
    <div
      className="home-datasets-search"
    >
      <form
        className="ccl-form search-form"
      >
        <input
          aria-label="Search datasets"
          className="ccl-text-input"
          id="home_search_datasets"
          name="homeSearchDatasets"
          placeholder="Search datasets"
          type="text"
        />
        <button
          aria-label="Search"
          className="ccl-button"
          type="submit"
        >
          <span
            className="ccl-icon-zoom"
          />
        </button>
      </form>
    </div>
    <div
      className="home-datasets-text"
    >
      <span>
        More than XXX.XXX datasets in our 
      </span>
      <a
        href="./dataset-catalogue.html"
      >
        Dataset catalogue
      </a>
    </div>
  </div>
</div>
`;
